src/vi/yurineko/src/eu/kanade/tachiyomi/extension/vi/yurineko/YuriNeko.kt

```diff
@@ -0,0 +1,406 @@
+package eu.kanade.tachiyomi.extension.vi.yurineko
+
+import eu.kanade.tachiyomi.extension.vi.yurineko.dto.ErrorResponseDto
+import eu.kanade.tachiyomi.extension.vi.yurineko.dto.MangaDto
+import eu.kanade.tachiyomi.extension.vi.yurineko.dto.MangaListDto
+import eu.kanade.tachiyomi.extension.vi.yurineko.dto.ReadResponseDto
+import eu.kanade.tachiyomi.extension.vi.yurineko.dto.UserDto
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservableSuccess
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.CacheControl
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import rx.Observable
+import java.net.URLDecoder
+import java.util.concurrent.TimeUnit
+import kotlin.math.ceil
+
+class YuriNeko : HttpSource() {
+
+    override val name = "YuriNeko"
+
+    override val baseUrl = "https://yurineko.net"
+
+    override val lang = "vi"
+
+    override val supportsLatest = false
+
+    private val apiUrl = "https://api.yurineko.net"
+
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimit(1, 2, TimeUnit.SECONDS)
+        .addInterceptor { authIntercept(it) }
+        .addInterceptor { chain ->
+            val response = chain.proceed(chain.request())
+
+            if (response.code >= 400 && response.body != null) {
+                val error = response.parseAs<ErrorResponseDto>()
+                response.close()
+                throw Exception("${error.message}\nĐăng nhập qua WebView và thử lại.")
```

Comment:
```suggestion
                throw IOException("${error.message}\nĐăng nhập qua WebView và thử lại.")
```

Interceptors only accepts `IOException`, otherwise it could cause crashes and unexpected behaviors.